source/plugins/ruby/kubelet_utils.rb

```diff
@@ -41,6 +41,55 @@ def get_node_capacity
+        cpu_allocatable = 1.0
+        memory_allocatable = 1.0
+
+        capacity_response = CAdvisorMetricsAPIClient.getAllMetricsCAdvisor(winNode: nil)
+        response = CAdvisorMetricsAPIClient.getCongifzCAdvisor(winNode: nil)
+        if !response.nil? && !response.body.nil?
+          kubereserved_cpu = JSON.parse(response.body)["kubeletconfig"]["kubeReserved"]["cpu"]
+          @log.info "get_node_allocatable::kubereserved_cpu  #{kubereserved_cpu}"
+          kubereserved_memory = JSON.parse(response.body)["kubeletconfig"]["kubeReserved"]["memory"]
+          @log.info "get_node_allocatable::kubereserved_memory #{kubereserved_memory}"
+          systemReserved_cpu = JSON.parse(response.body)["kubeletconfig"]["systemReserved"]["cpu"]
+          @log.info "get_node_allocatable::systemReserved_cpu  #{systemReserved_cpu}"
+          systemReserved_memory = JSON.parse(response.body)["kubeletconfig"]["systemReserved"]["memory"]
+          @log.info "get_node_allocatable::systemReserved_memory #{systemReserved_memory}"
+          evictionHard_cpu = JSON.parse(response.body)["kubeletconfig"]["evictionHard"]["nodefs.available"]
+          @log.info "get_node_allocatable::evictionHard_cpu #{evictionHard_cpu}"
+          evictionHard_memory = JSON.parse(response.body)["kubeletconfig"]["evictionHard"]["memory.available"]
+          @log.info "get_node_allocatable::evictionHard_memory #{evictionHard_memory}"
+
+          all_metrics = capacity_response.body.split("\n")
+          cpu_capacity = all_metrics.select { |m| m.start_with?("machine_cpu_cores") }.first.split.last.to_f * 1000
+          @log.info "get_node_allocatable::CPU Capacity #{cpu_capacity}"
+          memory_capacity_e = all_metrics.select { |m| m.start_with?("machine_memory_bytes") }.first.split.last
+          memory_capacity = BigDecimal(memory_capacity_e).to_f
+          @log.info "get_node_allocatable::Memory Capacity #{memory_capacity}"
+          cpu_capacity_number = cpu_capacity.to_i
+
+          # subtract to get allocatable. Formula : Allocatable = Capacity - ( kube reserved + system reserved + eviction threshold )
+          # https://kubernetes.io/docs/tasks/administer-cluster/reserve-compute-resources/#node-allocatable
+          cpu_allocatable  = cpu_capacity_number - ( kubereserved_cpu.tr('^0-9', '').to_i + systemReserved_cpu.tr('^0-9', '').to_i + ( evictionHard_cpu.tr('^0-9', '').to_i * cpu_capacity_number / 100 ) );
+          @log.info "CPU Allocatable #{cpu_allocatable}"
+          memory_allocatable = memory_capacity.to_i - ( ( kubereserved_memory.tr('^0-9', '').to_i * 1024 * 1024 ) + ( systemReserved_memory.tr('^0-9', '').to_i * 1024 * 1024 ) + ( evictionHard_memory.tr('^0-9', '').to_i * 1024 * 1024 ) );
+          @log.info "Memory Allocatable #{memory_allocatable}"
+
+          cpu_allocatable = BigDecimal(cpu_allocatable).to_f
+          memory_allocatable = BigDecimal(memory_allocatable).to_f
+
+          return [cpu_allocatable, memory_allocatable]
```

Comment:
https://kubernetes.io/docs/tasks/administer-cluster/reserve-compute-resources/#explicitly-reserved-cpu-list

the explicit cpuset provided by reserved-cpus will take precedence over the CPUs defined by --kube-reserved and --system-reserved options.